src/config/controller.rs

```diff
@@ -24,9 +25,46 @@ use super::midi::{self, ToMidiEvent};
 #[serde(tag = "kind", rename_all = "lowercase")]
 pub(super) enum Controller {
     Midi(MidiController),
+    Multi(HashMap<String, Controller>),
     Keyboard,
 }
 
+fn driver_from_midi_config(
+    config: &MidiController,
+    midi_device: Option<Arc<dyn crate::midi::Device>>,
+) -> Result<Arc<crate::controller::midi::Driver>, Box<dyn Error>> {
+    match midi_device {
+        Some(midi_device) => Ok(Arc::new(crate::controller::midi::Driver::new(
+            midi_device,
+            config.play.to_midi_event()?,
+            config.prev.to_midi_event()?,
+            config.next.to_midi_event()?,
+            config.stop.to_midi_event()?,
+            config.all_songs.to_midi_event()?,
+            config.playlist.to_midi_event()?,
+        ))),
+        None => Err("No MIDI device found for MIDI controller.".into()),
+    }
+}
+
+fn driver_from_midi_config2(
```

Comment:
Could we coalesce this and the other driver from midi config function into one somehow? If not, could we rename this?

Reply:
Thanks for the feedback! I intended this to be only one function but I could not get the compiler to agree yet 😆. I'll give it another shot, I don't like the current solution either.